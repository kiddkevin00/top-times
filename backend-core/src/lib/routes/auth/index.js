const Controller = require('../../controllers/auth.controller');
const authCheckMiddleware = require('../../middlewares/auth-check');
const { Router } = require('express');

const router = Router();

router.post('/signup', Controller.signup);
router.post('/login', Controller.login);
router.get('/logout', Controller.logout);
router.get('/check', [authCheckMiddleware], Controller.getMyProfile);

module.exports = exports = router;
