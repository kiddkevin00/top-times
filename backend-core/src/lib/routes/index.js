/**
 * This is the place for exposing module(s) for route component.
 */

const authRoutes = require('./auth/');
const usersRoutes = require('./users/');
const errorHandlerMiddleware = require('../middlewares/error-handler');
const constants = require('../constants/');
const packageJson = require('../../../package.json');
const errorHandler = require('errorhandler');
const { Router } = require('express');

const serverStartTimestamp = new Date();
const containerId = process.env.HOSTNAME;

function setupRoutes(app) {
  const { version } = packageJson;
  const majorVersion = version.slice(0, version.indexOf('.'));

  app.get('/ping', (req, res) =>
    res.json({
      uptimeInSec: (new Date().getTime() - serverStartTimestamp.getTime()) / 1000,
      hostname: containerId || 'N/A',
    })
  );
  app.get('/health', (req, res) =>
    res.json({
      version: packageJson.version,
      self: {
        name: packageJson.name,
        version: packageJson.version,
        status: constants.SYSTEM.HTTP_STATUS_CODES.OK,
        serverTimestamp: new Date().toString(),
        hostname: containerId || 'N/A',
      },
      dependencies: {
        http: [],
      },
    })
  );

  app.use(`/api/v${majorVersion}`, setupApiRoutes());

  // All not-found API endpoints should return a custom 404 page.
  app.route('/:url(api)/*').get((req, res) =>
    res.render('404', err => {
      if (err) {
        return res.status(constants.SYSTEM.HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR).json(err);
      }
      return res.status(constants.SYSTEM.HTTP_STATUS_CODES.NOT_FOUND).render('404');
    })
  );

  app.use(errorHandlerMiddleware);

  if (app.get('env') !== 'production') {
    app.use(errorHandler()); // Error handler - has to be the last.
  }

  return app;
}

function setupApiRoutes() {
  const router = Router();

  router.use('/auth', authRoutes);
  router.use('/users', usersRoutes);

  return router;
}

module.exports = exports = setupRoutes;
