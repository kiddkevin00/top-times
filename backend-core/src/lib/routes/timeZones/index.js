const Controller = require('../../controllers/timeZones.controller');
const authCheckMiddleware = require('../../middlewares/auth-check');
const { Router } = require('express');

const router = Router();

router.post('/', [authCheckMiddleware], Controller.addTimeZone);
router.get('/', [authCheckMiddleware], Controller.getTimeZone);
router.patch('/', [authCheckMiddleware], Controller.updateTimeZone);
router.delete('/', [authCheckMiddleware], Controller.removeTimeZone);

module.exports = exports = router;
