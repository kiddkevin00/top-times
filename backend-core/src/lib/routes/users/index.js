const Controller = require('../../controllers/users.controller');
const authCheckMiddleware = require('../../middlewares/auth-check');
const { Router } = require('express');

const router = Router();

router.get('/', [authCheckMiddleware], Controller.getUserProfile);
router.patch('/', [authCheckMiddleware], Controller.updateUserProfile);
router.delete('/', [authCheckMiddleware], Controller.suspendUser);

module.exports = exports = router;
