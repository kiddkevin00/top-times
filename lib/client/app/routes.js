import PageHeader from './containers/PageHeader';
import Home from './containers/Home';
import Login from './containers/Login';
import CreateAccount from './containers/CreateAccount';
import ForgotPassword from './containers/ForgotPassword';
import TimeManagement from './containers/TimeManagement';
import UserManagement from './containers/UserManagement';
import AccountInfo from './containers/AccountInfo';
import createStore from './store';
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';

const initialState = window.INITIAL_STATE;
const store = createStore(initialState);

const routes = (
  <Provider store={store}>
    <Router>
      <div>
        <PageHeader />

        <Switch>
          <Route exact={true} path="/home" component={Home} />
          <Route exact={true} path="/login" component={Login} />
          <Route exact={true} path="/create-account" component={CreateAccount} />
          <Route exact={true} path="/forgot-password" component={ForgotPassword} />
          <Route exact={true} path="/manage-timezones" component={TimeManagement} />
          <Route exact={true} path="/manage-timezones/:id" component={props => <TimeManagement {...props} />} />
          <Route exact={true} path="/manage-users" component={UserManagement} />
          <Route exact={true} path="/account-info" component={AccountInfo} />
          <Route exact={true} path="/account-info/:id" component={props => <AccountInfo {...props} />} />

          <Redirect from="/" to="/home" />
        </Switch>
      </div>
    </Router>
  </Provider>
);

export { routes as default };
