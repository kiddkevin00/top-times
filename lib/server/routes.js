const packageJson = require('../../package.json');
const errorHandler = require('errorhandler');
const path = require('path');

const serverStartTimestamp = new Date();
const containerId = process.env.HOSTNAME;

function setupRoutes(app) {
  app.get('/ping', (req, res) =>
    res.json({
      uptimeInSec: (new Date().getTime() - serverStartTimestamp.getTime()) / 1000,
      hostname: containerId || 'N/A',
    })
  );
  app.get('/health', (req, res) =>
    res.json({
      version: packageJson.version,
      self: {
        name: packageJson.name,
        version: packageJson.version,
        status: 200,
        serverDateStamp: new Date().toString(),
        hostname: containerId || 'N/A',
      },
      dependencies: {
        http: [],
      },
    })
  );

  // All not-found GET endpoints should return an custom 404 page (this should be sync with `CopyWebpackPlugin` in webpack.config.js).
  app.route('/:url(images|pdf|.well-known)/*').get((req, res) =>
    res.render('404', err => {
      if (err) {
        return res.status(500).json(err);
      }
      return res.sendStatus(500);
    })
  );

  // All other endpoints should redirect to the index.html.
  app.use((req, res) => {
    let headers;

    if (process.env.NODE_ENV === 'production') {
      headers = { 'Cache-Control': 'no-cache' };
    } else {
      headers = { 'Cache-Control': 'no-store' };
    }
    res.set(headers);

    return res.sendFile(path.resolve(__dirname, '../../build/', 'index.html'));
  });

  if (process.env.NODE_ENV !== 'production') {
    app.use(errorHandler()); // Error handler - has to be the last.
  }
}

module.exports = exports = setupRoutes;
