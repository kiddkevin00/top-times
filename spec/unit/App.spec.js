import routes from '../../lib/client/app/routes';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Routes element', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(routes);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
