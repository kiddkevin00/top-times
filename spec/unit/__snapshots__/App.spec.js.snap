// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes element matches snapshot 1`] = `
<BrowserRouter>
  <div>
    <Connect(withRouter(UnconnectedPageHeader)) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/home"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/create-account"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/forgot-password"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/manage-timezones"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/manage-timezones/:userId"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/manage-users"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/manage-timezone"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/account-info"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/account-info/:id"
      />
      <Redirect
        from="/"
        push={false}
        to="/home"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
